tests/src/Plugins.cpp

```diff
@@ -645,12 +645,19 @@ TEST_F(PluginTest, testTcpWebSocket)
 
     RequestWaiter waiter;
     std::vector<std::pair<bool, QByteArray>> messages;
+    bool open = false;
     lua->set("done", [&] {
         waiter.requestDone();
     });
     lua->set("add", [&](bool isText, QByteArray data) {
+        EXPECT_TRUE(open);
         messages.emplace_back(isText, std::move(data));
     });
+    // On GCC in release mode, using set() would cause the done function to be called instead.
+    lua->set_function("open", [&] {
```

Comment:
Don't know if it's us using sol incorrectly or GCC doing some weird optimization. It doesn't happen in debug mode, and doesn't happen on Clang and MSVC.

Reply:
Related to https://github.com/ThePhD/sol2/issues/608#issuecomment-372876206 https://sol2.readthedocs.io/en/latest/functions.html#binding-callable-objects

Also fixable by making the "open" function here accept a param, but using set_function is perfectly reasonable